node/src/accountant/scanners_utils.rs

```diff
@@ -127,6 +197,84 @@ pub mod payable_scanner_utils {
         })
     }
 
+    pub fn debugging_summary_after_error_separation(
+        oks: &[&PendingPayable],
+        errs_opt: &Option<PayableTransactingErrorEnum>,
+    ) -> String {
+        format!(
+            "Got {} properly sent payables of {} attempts",
+            oks.len(),
+            count_total_errors(errs_opt)
+                .map(|err_count| (err_count + oks.len()).to_string())
+                .unwrap_or_else(|| "an unknown number of".to_string())
+        )
+    }
+
+    pub(super) fn count_total_errors(
+        full_set_of_errors: &Option<PayableTransactingErrorEnum>,
+    ) -> Option<usize> {
+        match full_set_of_errors {
+            Some(errors) => match errors {
+                LocallyCausedError(blockchain_error) => match blockchain_error {
+                    PayablePaymentError::Sending { hashes, .. } => Some(hashes.len()),
+                    _ => None,
+                },
+                RemotelyCausedErrors(hashes) => Some(hashes.len()),
+            },
+            None => Some(0),
+        }
+    }
+
+    pub fn mark_pending_payable_fatal_error(
+        sent_payments: &[&PendingPayable],
+        nonexistent: &[RefWalletAndRowidOptCoupledWithHash],
+        error: PayableDaoError,
+        missing_fingerprints_msg_maker: fn(&[RefWalletAndRowidOptCoupledWithHash]) -> String,
+        logger: &Logger,
+    ) {
+        if !nonexistent.is_empty() {
+            error!(logger, "{}", missing_fingerprints_msg_maker(nonexistent))
+        };
+        panic!(
+            "Unable to create a mark in the payable table for wallets {} due to {:?}",
+            comma_joined_stringifiable(sent_payments, |pending_p| pending_p
+                .recipient_wallet
+                .to_string()),
+            error
+        )
+    }
+
+    pub fn err_msg_if_failed_without_existing_fingerprints(
+        nonexistent: VecOfRowidOptAndHash,
+        serialize_hashes: fn(&[H256]) -> String,
+    ) -> Option<String> {
+        let hashes_of_nonexistent = nonexistent
+            .iter()
+            .map(|(_, hash)| *hash)
+            .collect::<Vec<H256>>();
+        nonexistent.is_empty().not().then_some(format!(
+            "Ran into failed transactions {} with missing fingerprints. System no longer reliable",
+            serialize_hashes(&hashes_of_nonexistent),
+        ))
+    }
+
+    pub fn log_failed_payments_and_return_rowids_and_hashes(
+        ids_of_payments: VecOfRowidOptAndHash,
+        serialize_hashes: fn(&[H256]) -> String,
+        logger: &Logger,
+    ) -> (Vec<u64>, Vec<H256>) {
+        let (rowids, hashes): (Vec<u64>, Vec<H256>) = ids_of_payments
+            .into_iter()
+            .map(|(ever_some_rowid, hash)| (ever_some_rowid.expectv("validated rowid"), hash))
+            .unzip();
+        warning!(
```

Comment:
Forensics would be a little easier if this log were placed closer to the actual delete machinery.